Gems/Profiler/Code/Source/CpuProfilerImpl.h

```diff
@@ -124,15 +124,22 @@ namespace Profiler
 
     private:
         static constexpr AZStd::size_t MaxFramesToSave = 2 * 60 * 120; // 2 minutes of 120fps
-        static constexpr AZStd::size_t MaxRegionStringPoolSize = 16384; // Max amount of unique strings to save in the pool before throwing warnings.
+        static constexpr AZStd::size_t MaxDynamicStringPoolSize = 4096; // Max amount of unique strings to save in the pool before throwing warnings.
 
         // Lazily create and register the local thread data
         void RegisterThreadStorage();
 
+        const char* GetOrCreateFormatedString(const char* format, size_t argsCount, va_list args);
+
         // ThreadId -> ThreadTimeRegionMap
         // On the start of each frame, this map will be updated with the last frame's profiling data.
         TimeRegionMap m_timeRegionMap;
 
+        // String pool for storing region names submitted at runtime. Each call to any AZ_PROFILE_* macro with format string arguments
+        // will either construct a string in this pool or use an already-existing entry.
+        AZStd::unordered_map<AZStd::size_t, AZStd::string> m_dynamicNameStringPool;
```

Comment:
I'm curious if it's worth actually trying to cache dynamic strings, given that these strings will (I assume) often be changing every frame.

For example, even a profile event like `Frame 209` will be changing every frame.

Reply:
It is and it isn't.  Currently, we don't have any active markers like that but is not outside the realm of possibility.  You are 100% right something like "Frame %d" would absolutely destroy a caching system.  That said, always performing the format is at a higher cost, even now, compared to caching them.  

Not that debug is the greatest profile environment but with caching the hit is about 3-4 fps (average of ~32 down to ~28 fps) while always running the format is 3-4x that at ~12 fps hit (20 to sub-20 fps).  Now in profile that hit is certainly less noticeable but still there; an unstable 1-3 fps hit for always formatting (likely from memory thrash) vs a more consistent ~1 fps hit when caching.

Reply:
There are also use cases that are not as dynamic. For instance in writing a drive letter in a marker like `StorageDrive C:`. These are dynamic, but only setup once.